diff --git a/lib/commonjs/hooks/useBoundingClientRect.js b/lib/commonjs/hooks/useBoundingClientRect.js
index b4a90b76ee55bf2cad9cf461017621b1ddab0fe1..3140ff9d1d3bd9ae28fc5124ac642af0eda74ea7 100644
--- a/lib/commonjs/hooks/useBoundingClientRect.js
+++ b/lib/commonjs/hooks/useBoundingClientRect.js
@@ -45,19 +45,25 @@ function useBoundingClientRect(ref, handler) {
       return;
     }
 
+    if (
     // @ts-ignore 👉 https://github.com/facebook/react/commit/53b1f69ba
-    if (ref.current.unstable_getBoundingClientRect !== null) {
+    ref.current.unstable_getBoundingClientRect !== null &&
+    // @ts-ignore 👉 https://github.com/facebook/react/commit/53b1f69ba
+    typeof ref.current.unstable_getBoundingClientRect === 'function') {
       // @ts-ignore https://github.com/facebook/react/commit/53b1f69ba
-      const layout = ref.current.unstable_getBoundingClientRect();
+      var layout = ref.current.unstable_getBoundingClientRect();
       handler(layout);
       return;
     }
 
+    if (
     // @ts-ignore once it `unstable_getBoundingClientRect` gets stable 🤞.
-    if (ref.current.getBoundingClientRect !== null) {
+    ref.current.getBoundingClientRect !== null &&
+    // @ts-ignore once it `unstable_getBoundingClientRect` gets stable.
+    typeof ref.current.getBoundingClientRect === 'function') {
       // @ts-ignore once it `unstable_getBoundingClientRect` gets stable.
-      const layout = ref.current.getBoundingClientRect();
-      handler(layout);
+      var _layout = ref.current.getBoundingClientRect();
+      handler(_layout);
     }
   });
 }
diff --git a/lib/module/hooks/useBoundingClientRect.js b/lib/module/hooks/useBoundingClientRect.js
index a723aede9d4cfbb46f5985c531e0dae8f517aba8..2da7edb539836fef15f9ed29d38cfe4608afd121 100644
--- a/lib/module/hooks/useBoundingClientRect.js
+++ b/lib/module/hooks/useBoundingClientRect.js
@@ -41,16 +41,22 @@ export function useBoundingClientRect(ref, handler) {
       return;
     }
 
+    if (
     // @ts-ignore 👉 https://github.com/facebook/react/commit/53b1f69ba
-    if (ref.current.unstable_getBoundingClientRect !== null) {
+    ref.current.unstable_getBoundingClientRect !== null &&
+    // @ts-ignore 👉 https://github.com/facebook/react/commit/53b1f69ba
+    typeof ref.current.unstable_getBoundingClientRect === 'function') {
       // @ts-ignore https://github.com/facebook/react/commit/53b1f69ba
       const layout = ref.current.unstable_getBoundingClientRect();
       handler(layout);
       return;
     }
 
+    if (
     // @ts-ignore once it `unstable_getBoundingClientRect` gets stable 🤞.
-    if (ref.current.getBoundingClientRect !== null) {
+    ref.current.getBoundingClientRect !== null &&
+    // @ts-ignore once it `unstable_getBoundingClientRect` gets stable.
+    typeof ref.current.getBoundingClientRect === 'function') {
       // @ts-ignore once it `unstable_getBoundingClientRect` gets stable.
       const layout = ref.current.getBoundingClientRect();
       handler(layout);
diff --git a/src/hooks/useBoundingClientRect.ts b/src/hooks/useBoundingClientRect.ts
index cc85c8ced2de8ec514360368ed20af733f8f9aec..9abe8294d6004be4500871e46a3621a9e5b9d93b 100644
--- a/src/hooks/useBoundingClientRect.ts
+++ b/src/hooks/useBoundingClientRect.ts
@@ -55,16 +55,24 @@ export function useBoundingClientRect(
       return;
     }
 
-    // @ts-ignore 👉 https://github.com/facebook/react/commit/53b1f69ba
-    if (ref.current.unstable_getBoundingClientRect !== null) {
+    if (
+      // @ts-ignore 👉 https://github.com/facebook/react/commit/53b1f69ba
+      ref.current.unstable_getBoundingClientRect !== null &&
+      // @ts-ignore 👉 https://github.com/facebook/react/commit/53b1f69ba
+      typeof ref.current.unstable_getBoundingClientRect === 'function'
+    ) {
       // @ts-ignore https://github.com/facebook/react/commit/53b1f69ba
       const layout = ref.current.unstable_getBoundingClientRect();
       handler(layout);
       return;
     }
 
-    // @ts-ignore once it `unstable_getBoundingClientRect` gets stable 🤞.
-    if (ref.current.getBoundingClientRect !== null) {
+    if (
+      // @ts-ignore once it `unstable_getBoundingClientRect` gets stable.
+      ref.current.getBoundingClientRect !== null &&
+      // @ts-ignore once it `unstable_getBoundingClientRect` gets stable.
+      typeof ref.current.getBoundingClientRect === 'function'
+    ) {
       // @ts-ignore once it `unstable_getBoundingClientRect` gets stable.
       const layout = ref.current.getBoundingClientRect();
       handler(layout);
