diff --git a/android/build.gradle b/android/build.gradle
index 72eb7862a2530439bd7227adc0649fcfa33fcb80..7f673a4731cdb427fc5da2fb120a58c90d01686b 100755
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -85,7 +85,7 @@ repositories {
 }
 
 dependencies {
-    implementation "org.jetbrains.kotlin:kotlin-stdlib:${safeExtGet('kotlin_stdlib_version', '2.4.10')}"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:${safeExtGet('kotlin_stdlib_version', safeExtGet('kotlinVersion', '2.4.10'))}"
     implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
     implementation "com.android.installreferrer:installreferrer:${safeExtGet('installReferrerVersion', '2.2')}"
     if (includeConnector){
diff --git a/index.ts b/index.ts
index 8500ac7746987997d16be5f47c2b1ea77cdef4b1..87c3fc9da54e205ed38d64994450463558dcd028 100644
--- a/index.ts
+++ b/index.ts
@@ -54,7 +54,7 @@ export type SubscriptionPurchaseEventDataSource = PurchaseEventDataSourceBase;
 export type InAppPurchaseEventDataSource = PurchaseEventDataSourceBase;
 
 export interface PurchaseConnector {
-  create(config: PurchaseConnectorConfig): void;
+  create(config: PurchaseConnectorConfig): Promise<void>;
   startObservingTransactions(): void;
   stopObservingTransactions(): void;
 
@@ -257,7 +257,7 @@ AppsFlyerPurchaseConnector.create = (config: PurchaseConnectorConfig) => {
   if (!config) {
     throw new MissingConfigurationException();
   }
-  PCAppsFlyer.create(config);
+  return PCAppsFlyer.create(config);
 };
 
 export { AppsFlyerPurchaseConnector };
