diff --git a/src/optionalDependencies/getPhotos.ts b/src/optionalDependencies/getPhotos.ts
index c0bbe84696a35ae1004aae3ca9b371815f2998c6..e4f008bcf2a816fa846f2c8e17f0579f2da33f8b 100644
--- a/src/optionalDependencies/getPhotos.ts
+++ b/src/optionalDependencies/getPhotos.ts
@@ -63,7 +63,7 @@ export const getPhotos = MediaLibrary
               height: asset.height,
               name: asset.filename,
               size: 0,
-              thumb_url: asset.mediaType === 'photo' ? undefined : asset.uri,
+              thumb_url: asset.mediaType === 'photo' ? undefined : (localUri || asset.uri),
               type: mimeType,
               uri: localUri || asset.uri,
               width: asset.width,
